dm/master/server.go

```diff
@@ -1483,15 +1587,14 @@ func (s *Server) UpdateMasterConfig(ctx context.Context, req *pb.UpdateMasterCon
 	// add new worker
 	for _, workerAddr := range cfg.DeployMap {
 		if _, ok := s.workerClients[workerAddr]; !ok {
-			conn, err2 := grpc.Dial(workerAddr, grpc.WithInsecure(), grpc.WithBackoffMaxDelay(3*time.Second))
-			if err2 != nil {
+			s.workerClients[workerAddr], err = workerrpc.NewGRPCClient(workerAddr)
```

Comment:
if `err != nil`, a `nil` client will added into `workerClients`?

Reply:
fixed in https://github.com/pingcap/dm/pull/157/commits/6c0d022d6617653a0020d2bc91e80433d143e860